[Bug]: panic in mtmOrFundingSettlement

### Problem encountered

When adding a new scenario to market-sim for testing new liquidation strategies, core panics.

vega-market-sim dump including replay file:
[vega-sim-dbocf8nm.zip](https://github.com/vegaprotocol/vega/files/13418684/vega-sim-dbocf8nm.zip)

```
panic: No win transfers, settlement balance non-zero

goroutine 3099 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x40?, 0x1?, {0x1?, 0x0?, 0x0?})
	/Users/caed/go/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/entry.go:198 +0x7c
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0x1400f39a680, {0x1400fd85640, 0x1, 0x1})
	/Users/caed/go/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/entry.go:264 +0x2bc
go.uber.org/zap.(*Logger).Panic(0xd75b1ff1cf432e00?, {0x10716eff2?, 0x0?}, {0x1400fd85640, 0x1, 0x1})
	/Users/caed/go/pkg/mod/go.uber.org/zap@v1.24.0/logger.go:258 +0x54
code.vegaprotocol.io/vega/core/collateral.(*Engine).mtmOrFundingSettlement(0x140003fe620, {0x10838a758, 0x1400f3dd8c0}, {0x1400b257fc0, 0x40}, {0x1400f3f6500, 0x2, 0x13}, {0x1400b257bc0, 0x40}, ...)
	/Users/caed/vega-market-sim/extern/vega/core/collateral/engine.go:1468 +0x1eb4
code.vegaprotocol.io/vega/core/collateral.(*Engine).MarkToMarket(0x1400fd97600?, {0x10838a758?, 0x1400f3dd8c0?}, {0x1400b257fc0?, 0x1?}, {0x1400f3f6500?, 0x1400fdad960?, 0x20?}, {0x1400b257bc0?, 0x1400fdb14d0?})
	/Users/caed/vega-market-sim/extern/vega/core/collateral/engine.go:1315 +0x38
code.vegaprotocol.io/vega/core/execution/future.(*Market).collateralAndRisk(0x140005b2dc0, {0x10838a758, 0x1400f3dd8c0}, {0x1400f3f6500, 0x2, 0x13})
	/Users/caed/vega-market-sim/extern/vega/core/execution/future/market.go:2626 +0x15c
code.vegaprotocol.io/vega/core/execution/future.(*Market).confirmMTM(0x140005b2dc0, {0x10838a758, 0x1400f3dd8c0}, 0x0)
	/Users/caed/vega-market-sim/extern/vega/core/execution/future/market.go:2297 +0x2c8
code.vegaprotocol.io/vega/core/execution/future.(*Market).checkNetwork(0x140005b2dc0, {0x10838a758, 0x1400f3dd8c0}, {0x1400fd8bb40?, 0x1400cee0b38?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/execution/future/liquidation.go:90 +0x38c
code.vegaprotocol.io/vega/core/execution/future.(*Market).OnTick(0x140005b2dc0, {0x10838a758, 0x1400f3dd8c0}, {0x40?, 0x0?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/execution/future/market.go:889 +0x2a0
code.vegaprotocol.io/vega/core/execution.(*Engine).OnTick(0x140002e2580, {0x10838a758, 0x1400f3dd8c0}, {0x1?, 0x1083a5450?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/execution/engine.go:1302 +0x398
code.vegaprotocol.io/vega/core/vegatime.(*Svc).notify(...)
	/Users/caed/vega-market-sim/extern/vega/core/vegatime/service.go:107
code.vegaprotocol.io/vega/core/vegatime.(*Svc).SetTimeNow(0x14000298f00, {0x10838a758, 0x1400f3dd8c0}, {0x40?, 0x1400fd840c0?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/vegatime/service.go:80 +0x26c
code.vegaprotocol.io/vega/core/processor.(*App).OnBeginBlock(0x14000c3aa80, 0x18c, {0x1400fd8b840, 0x40}, {0x1400fd8b840?, 0x0?, 0x0?}, {0x0, 0x0}, {0x1400fd840c0?, ...})
	/Users/caed/vega-market-sim/extern/vega/core/processor/abci.go:963 +0x794
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).FinalizeBlock(0x14000dd0240, {0x108169f40?, 0x1?}, 0x1400db25500)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/abci/abci.go:162 +0x330
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).FinalizeBlock(0x1400e2f11e8?, {0x10838a6e8?, 0x140000780d8?}, 0x1400d836240?)
	/Users/caed/vega-market-sim/extern/vega/cmd/vega/commands/node/app_wrapper.go:91 +0x30
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0x140004ce000)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:198 +0x448
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0x140004ce000, 0x3b9aca00)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:275 +0x198
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:132 +0x2c
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:131 +0x1ec
```

### Observed behaviour

panic

### Expected behaviour

no panic

### Steps to reproduce

```bash
1. replay using replay file
```


### Software version

7f99ce075e5ae3ab2d7004f76a52463efc63ee8c

### Failing test

_No response_

### Jenkins run

_No response_

### Configuration used

```shell
{
  "genesis_time": "2023-05-09T10:00:00.000000000Z",
  "chain_id": "CUSTOM",
  "initial_height": "0",
  "consensus_params": {
    "block": {
      "max_bytes": "22020096",
      "max_gas": "-1",
      "time_iota_ms": "1"
    },
    "evidence": {
      "max_age_num_blocks": "100000",
      "max_age_duration": "172800000000000",
      "max_bytes": "1048576"
    },
    "validator": {
      "pub_key_types": ["ed25519"]
    },
    "version": {}
  },
  "validators": [
    {
      "address": "7DFD3744A773C166B36F170E4F5EF3E998703F4E",
      "pub_key": {
        "type": "tendermint/PubKeyEd25519",
        "value": "2lPO6I0p5uCBZzeoA2+ECJ2+57LtLVMjXpxH5za9TvE="
      },
      "power": "10",
      "name": ""
    }
  ],
  "app_hash": "",
  "app_state": {
    "assets": {
      "VOTE": {
        "name": "VOTE",
        "symbol": "VOTE",
        "decimals": 5,
        "min_lp_stake": "1",
        "source": {
          "builtin_asset": {
            "max_faucet_amount_mint": "100000000000"
          }
        }
      }
    },
    "validators": {
      "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=": {
        "id": "c2b0ab69fb1bd19862c04df401920ea62d3e0bce41a50148240f42017824d17a",
        "vega_pub_key": "dfcf0fa8d67cfa11b3a511e320f3c47807a5e0735cc06dd9c2bc00f7ccc67117",
        "vega_pub_key_index": 1,
        "ethereum_address": "0x9e5BEEC6E56B28cCbd02864840B0f1e0125e42Ce",
        "tm_pub_key": "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=",
        "info_url": "",
        "country": "",
        "name": "",
        "avatar_url": ""
      }
    },
    "network": {
      "replay_attack_threshold": 150
    },
    "network_parameters": {
      "blockchains.ethereumConfig": "{\"network_id\": \"3\", \"chain_id\": \"3\", \"collateral_bridge_contract\": { \"address\": \"0xa6F1E140daC13002Dfd9789D6dBA59117c717D7a\" }, \"confirmations\": 50, \"staking_bridge_contract\": { \"address\": \"0xfce2CC92203A266a9C8e67461ae5067c78f67235\", \"deployment_block_height\": 11001702}, \"multisig_control_contract\": {\"address\": \"0xCF6d41235911184fe6F35D47207813bFF3B91601\", \"deployment_block_height\": 12710009 } }",
      "governance.proposal.asset.maxClose": "8760h0m0s",
      "governance.proposal.asset.maxEnact": "8760h0m0s",
      "governance.proposal.asset.minClose": "1s",
      "governance.proposal.asset.minEnact": "1s",
      "governance.proposal.asset.minProposerBalance": "1",
      "governance.proposal.asset.minVoterBalance": "1",
      "governance.proposal.asset.requiredMajority": "0.66",
      "governance.proposal.asset.requiredParticipation": "0.00001",
      "governance.proposal.freeform.maxClose": "8760h0m0s",
      "governance.proposal.freeform.minClose": "1s",
      "governance.proposal.freeform.minProposerBalance": "1",
      "governance.proposal.freeform.minVoterBalance": "1",
      "governance.proposal.freeform.requiredMajority": "0.66",
      "governance.proposal.freeform.requiredParticipation": "0.00001",
      "governance.proposal.market.maxClose": "8760h0m0s",
      "governance.proposal.market.maxEnact": "8760h0m0s",
      "governance.proposal.market.minClose": "1s",
      "governance.proposal.market.minEnact": "2s",
      "governance.proposal.market.minProposerBalance": "1",
      "governance.proposal.market.minVoterBalance": "1",
      "governance.proposal.market.requiredMajority": "0.66",
      "governance.proposal.market.requiredParticipation": "0.00001",
      "governance.proposal.updateAsset.maxClose": "720h",
      "governance.proposal.updateAsset.maxEnact": "720h",
      "governance.proposal.updateAsset.minClose": "1s",
      "governance.proposal.updateAsset.minEnact": "1s",
      "governance.proposal.updateAsset.minProposerBalance": "1",
      "governance.proposal.updateAsset.minVoterBalance": "1",
      "governance.proposal.updateAsset.requiredMajority": "0.66",
      "governance.proposal.updateAsset.requiredParticipation": "0.09",
      "governance.proposal.updateMarket.maxClose": "8760h0m0s",
      "governance.proposal.updateMarket.maxEnact": "8760h0m0s",
      "governance.proposal.updateMarket.minClose": "1s",
      "governance.proposal.updateMarket.minEnact": "1s",
      "governance.proposal.updateMarket.minProposerBalance": "1",
      "governance.proposal.updateMarket.minProposerEquityLikeShare": "0",
      "governance.proposal.updateMarket.minVoterBalance": "1",
      "governance.proposal.updateMarket.requiredMajority": "0.66",
      "governance.proposal.updateMarket.requiredMajorityLP": "0.66",
      "governance.proposal.updateMarket.requiredParticipation": "0.00001",
      "governance.proposal.updateMarket.requiredParticipationLP": "0",
      "governance.proposal.updateNetParam.maxClose": "8760h0m0s",
      "governance.proposal.updateNetParam.maxEnact": "8760h0m0s",
      "governance.proposal.updateNetParam.minClose": "1s",
      "governance.proposal.updateNetParam.minEnact": "1s",
      "governance.proposal.updateNetParam.minProposerBalance": "1",
      "governance.proposal.updateNetParam.minVoterBalance": "1",
      "governance.proposal.updateNetParam.requiredMajority": "0.5",
      "governance.proposal.updateNetParam.requiredParticipation": "0.00001",
      "governance.proposal.referralProgram.maxClose": "8760h0m0s",
      "governance.proposal.referralProgram.maxEnact": "8760h0m0s",
      "governance.proposal.referralProgram.minClose": "1s",
      "governance.proposal.referralProgram.minEnact": "1s",
      "governance.proposal.referralProgram.minProposerBalance": "1",
      "governance.proposal.referralProgram.minVoterBalance": "1",
      "governance.proposal.referralProgram.requiredMajority": "0.5",
      "governance.proposal.referralProgram.requiredParticipation": "0.00001",
      "governance.proposal.VolumeDiscountProgram.maxClose": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.maxEnact": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.minClose": "1s",
      "governance.proposal.VolumeDiscountProgram.minEnact": "1s",
      "governance.proposal.VolumeDiscountProgram.minProposerBalance": "1",
      "governance.proposal.VolumeDiscountProgram.minVoterBalance": "1",
      "governance.proposal.VolumeDiscountProgram.requiredMajority": "0.5",
      "governance.proposal.VolumeDiscountProgram.requiredParticipation": "0.00001",
      "governance.proposal.transfer.maxClose": "8760h0m0s",
      "governance.proposal.transfer.maxEnact": "8760h0m0s",
      "governance.proposal.transfer.minClose": "1s",
      "governance.proposal.transfer.minEnact": "1s",
      "governance.proposal.transfer.minProposerBalance": "1",
      "governance.proposal.transfer.minVoterBalance": "1",
      "governance.proposal.transfer.requiredMajority": "0.5",
      "governance.proposal.transfer.requiredParticipation": "0.00001",
      "limits.markets.maxPeggedOrders": "1500",
      "limits.markets.proposePerpetualEnabled": "1",
      "market.auction.maximumDuration": "168h",
      "market.auction.minimumDuration": "1s",
      "market.fee.factors.infrastructureFee": "0.0005",
      "market.fee.factors.makerFee": "0.0002",
      "market.liquidity.bondPenaltyParameter": "0.1",
      "market.liquidity.earlyExitPenalty": "0.1",
      "market.liquidity.maximumLiquidityFeeFactorLevel": "0.03",
      "market.liquidity.minimum.probabilityOfTrading.lpOrders": "1e-6",
      "market.liquidity.probabilityOfTrading.tau.scaling": "10.0",
      "market.liquidity.providersFeeCalculationTimeStep": "1s",
      "market.liquidity.sla.nonPerformanceBondPenaltyMax": "0.5",
      "market.liquidity.sla.nonPerformanceBondPenaltySlope": "2",
      "market.liquidity.stakeToCcyVolume": "1.0",
      "market.liquidity.targetstake.triggering.ratio": "0.25",
      "market.liquidityProvision.minLpStakeQuantumMultiple": "5000",
      "market.liquidityProvision.shapes.maxSize": "100",
      "market.margin.scalingFactors": "{\"search_level\": 1.1, \"initial_margin\": 1.5, \"collateral_release\": 1.7}",
      "market.monitor.price.defaultParameters": "{\"triggers\": [{\"auction_extension\": 300, \"horizon\": 43200, \"probability\": \"0.9999999\"}] }",
      "market.stake.target.scalingFactor": "0.0001",
      "market.stake.target.timeWindow": "1h",
      "market.value.windowLength": "2h0m0s",
      "network.checkpoint.timeElapsedBetweenCheckpoints": "5m",
      "network.floatingPointUpdates.delay": "5m",
      "network.markPriceUpdateMaximumFrequency": "5s",
      "network.transaction.defaultgas": "1",
      "network.transactions.maxgasperblock": "3000000",
      "network.transactions.minBlockCapacity": "32",
      "network.validators.ersatz.multipleOfTendermintValidators": "0",
      "network.validators.ersatz.rewardFactor": "0.85",
      "network.validators.incumbentBonus": "0.05",
      "network.validators.minimumEthereumEventsForNewValidator": "3",
      "network.validators.multisig.numberOfSigners": "13",
      "network.validators.tendermint.number": "13",
      "reward.asset": "VOTE",
      "reward.staking.delegation.competitionLevel": "1.3",
      "reward.staking.delegation.delegatorShare": "0.8",
      "reward.staking.delegation.maxPayoutPerEpoch": "9890000000000000000000",
      "reward.staking.delegation.maxPayoutPerParticipant": "700000000000000000000",
      "reward.staking.delegation.minimumValidatorStake": "3000000000000000000000",
      "reward.staking.delegation.minValidators": "5",
      "reward.staking.delegation.optimalStakeMultiplier": "3.0",
      "reward.staking.delegation.payoutDelay": "0h",
      "reward.staking.delegation.payoutFraction": "1.0",
      "rewards.marketCreationQuantumMultiple": "1",
      "snapshot.interval.length": "300",
      "spam.pow.difficulty": "15",
      "spam.pow.hashFunction": "sha3_24_rounds",
      "spam.pow.increaseDifficulty": "0",
      "spam.pow.numberOfPastBlocks": "100",
      "spam.pow.numberOfTxPerBlock": "1000",
      "spam.protection.delegation.min.tokens": "100000000000000000",
      "spam.protection.max.batchSize": "30",
      "spam.protection.max.delegations": "360",
      "spam.protection.max.proposals": "300",
      "spam.protection.max.votes": "300",
      "spam.protection.maxUserTransfersPerEpoch": "10000",
      "spam.protection.minimumWithdrawalQuantumMultiple": "10",
      "spam.protection.minMultisigUpdates": "100000000000000000000",
      "spam.protection.proposal.min.tokens": "2000000000000000000000",
      "spam.protection.voting.min.tokens": "1000000000000000000",
      "spam.protection.applyReferral.min.funds": "0",
      "transfer.fee.factor": "0.001",
      "transfer.minTransferQuantumMultiple": "100",
      "validator.performance.scaling.factor": "0",
      "validators.delegation.minAmount": "1",
      "validators.epoch.length": "2m",
      "validators.vote.required": "0.67",
      "referralProgram.maxReferralTiers": "10",
      "referralProgram.maxReferralRewardFactor": "1",
      "referralProgram.maxReferralDiscountFactor": "1",
      "referralProgram.maxPartyNotionalVolumeByQuantumPerEpoch": "10000",
      "referralProgram.minStakedVegaTokens": "0",
      "referralProgram.maxReferralRewardProportion": "1"
    },
    "network_limits": {
      "propose_market_enabled": true,
      "propose_asset_enabled": true,
      "bootstrap_block_count": 0
    },
    "checkpoint": {
      "load_hash": ""
    }
  }
}
```


### Relevant log output

```shell
panic: No win transfers, settlement balance non-zero

goroutine 3099 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x40?, 0x1?, {0x1?, 0x0?, 0x0?})
	/Users/caed/go/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/entry.go:198 +0x7c
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0x1400f39a680, {0x1400fd85640, 0x1, 0x1})
	/Users/caed/go/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/entry.go:264 +0x2bc
go.uber.org/zap.(*Logger).Panic(0xd75b1ff1cf432e00?, {0x10716eff2?, 0x0?}, {0x1400fd85640, 0x1, 0x1})
	/Users/caed/go/pkg/mod/go.uber.org/zap@v1.24.0/logger.go:258 +0x54
code.vegaprotocol.io/vega/core/collateral.(*Engine).mtmOrFundingSettlement(0x140003fe620, {0x10838a758, 0x1400f3dd8c0}, {0x1400b257fc0, 0x40}, {0x1400f3f6500, 0x2, 0x13}, {0x1400b257bc0, 0x40}, ...)
	/Users/caed/vega-market-sim/extern/vega/core/collateral/engine.go:1468 +0x1eb4
code.vegaprotocol.io/vega/core/collateral.(*Engine).MarkToMarket(0x1400fd97600?, {0x10838a758?, 0x1400f3dd8c0?}, {0x1400b257fc0?, 0x1?}, {0x1400f3f6500?, 0x1400fdad960?, 0x20?}, {0x1400b257bc0?, 0x1400fdb14d0?})
	/Users/caed/vega-market-sim/extern/vega/core/collateral/engine.go:1315 +0x38
code.vegaprotocol.io/vega/core/execution/future.(*Market).collateralAndRisk(0x140005b2dc0, {0x10838a758, 0x1400f3dd8c0}, {0x1400f3f6500, 0x2, 0x13})
	/Users/caed/vega-market-sim/extern/vega/core/execution/future/market.go:2626 +0x15c
code.vegaprotocol.io/vega/core/execution/future.(*Market).confirmMTM(0x140005b2dc0, {0x10838a758, 0x1400f3dd8c0}, 0x0)
	/Users/caed/vega-market-sim/extern/vega/core/execution/future/market.go:2297 +0x2c8
code.vegaprotocol.io/vega/core/execution/future.(*Market).checkNetwork(0x140005b2dc0, {0x10838a758, 0x1400f3dd8c0}, {0x1400fd8bb40?, 0x1400cee0b38?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/execution/future/liquidation.go:90 +0x38c
code.vegaprotocol.io/vega/core/execution/future.(*Market).OnTick(0x140005b2dc0, {0x10838a758, 0x1400f3dd8c0}, {0x40?, 0x0?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/execution/future/market.go:889 +0x2a0
code.vegaprotocol.io/vega/core/execution.(*Engine).OnTick(0x140002e2580, {0x10838a758, 0x1400f3dd8c0}, {0x1?, 0x1083a5450?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/execution/engine.go:1302 +0x398
code.vegaprotocol.io/vega/core/vegatime.(*Svc).notify(...)
	/Users/caed/vega-market-sim/extern/vega/core/vegatime/service.go:107
code.vegaprotocol.io/vega/core/vegatime.(*Svc).SetTimeNow(0x14000298f00, {0x10838a758, 0x1400f3dd8c0}, {0x40?, 0x1400fd840c0?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/vegatime/service.go:80 +0x26c
code.vegaprotocol.io/vega/core/processor.(*App).OnBeginBlock(0x14000c3aa80, 0x18c, {0x1400fd8b840, 0x40}, {0x1400fd8b840?, 0x0?, 0x0?}, {0x0, 0x0}, {0x1400fd840c0?, ...})
	/Users/caed/vega-market-sim/extern/vega/core/processor/abci.go:963 +0x794
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).FinalizeBlock(0x14000dd0240, {0x108169f40?, 0x1?}, 0x1400db25500)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/abci/abci.go:162 +0x330
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).FinalizeBlock(0x1400e2f11e8?, {0x10838a6e8?, 0x140000780d8?}, 0x1400d836240?)
	/Users/caed/vega-market-sim/extern/vega/cmd/vega/commands/node/app_wrapper.go:91 +0x30
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0x140004ce000)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:198 +0x448
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0x140004ce000, 0x3b9aca00)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:275 +0x198
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:132 +0x2c
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:131 +0x1ec
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -48,6 +48,7 @@
 - [10132](https://github.com/vegaprotocol/vega/issues/10132) - Add mapping in `GraphQL` for update perps market proposal.
 - [10125](https://github.com/vegaprotocol/vega/issues/10125) - Wire the `JoinTeam` command in the wallet.
 - [10127](https://github.com/vegaprotocol/vega/issues/10127) - Untangle `ApplyReferralCode` and `JoinTeam` command verification.
+- [10147](https://github.com/vegaprotocol/vega/issues/10147) - Add network transfer largest share to the transfers if needed.
 
 ## 0.73.0
 

--- a/core/collateral/engine.go
+++ b/core/collateral/engine.go
@@ -1370,9 +1370,7 @@ func (e *Engine) mtmOrFundingSettlement(ctx context.Context, marketID string, tr
 		if transfer == nil {
 			// no error when getting MTM accounts, and no margin account == network position
 			// we are not interested in this event, continue here
-			if party != types.NetworkParty {
-				marginEvts = append(marginEvts, marginEvt)
-			}
+			marginEvts = append(marginEvts, marginEvt)
 			continue
 		}
 
@@ -2668,40 +2666,45 @@ func (e *Engine) getTransferRequest(p *types.Transfer, settle, insurance *types.
 			Type:  p.Type,
 		}
 	)
-	if p.Type == types.TransferTypeMTMLoss ||
-		p.Type == types.TransferTypePerpFundingLoss ||
-		p.Type == types.TransferTypeWin ||
-		p.Type == types.TransferTypeMarginLow {
-		// we do not care about errors here as the bond account is not mandatory for the transfers
-		// a partry would have a bond account only if it was also a market maker
-		mEvt.bond, _ = e.GetAccountByID(e.accountID(settle.MarketID, p.Owner, asset, types.AccountTypeBond))
-	}
-	if settle != nil && mEvt.margin == nil && p.Owner != types.NetworkParty {
-		// the accounts for the party we need
-		// the accounts for the trader we need
-		mEvt.margin, err = e.GetAccountByID(e.accountID(settle.MarketID, p.Owner, asset, types.AccountTypeMargin))
-		if err != nil {
-			e.log.Error(
-				"Failed to get the party margin account",
-				logging.String("owner-id", p.Owner),
-				logging.String("market-id", settle.MarketID),
-				logging.Error(err),
-			)
-			return nil, err
+	if p.Owner != types.NetworkParty {
+		if p.Type == types.TransferTypeMTMLoss ||
+			p.Type == types.TransferTypePerpFundingLoss ||
+			p.Type == types.TransferTypeWin ||
+			p.Type == types.TransferTypeMarginLow {
+			// we do not care about errors here as the bond account is not mandatory for the transfers
+			// a partry would have a bond account only if it was also a market maker
+			mEvt.bond, _ = e.GetAccountByID(e.accountID(settle.MarketID, p.Owner, asset, types.AccountTypeBond))
 		}
-	}
-	// we'll need this account for all transfer types anyway (settlements, margin-risk updates)
-	if mEvt.general == nil && p.Owner != types.NetworkParty {
-		mEvt.general, err = e.GetAccountByID(e.accountID(noMarket, p.Owner, asset, types.AccountTypeGeneral))
-		if err != nil {
-			e.log.Error(
-				"Failed to get the party general account",
-				logging.String("owner-id", p.Owner),
-				logging.String("market-id", settle.MarketID),
-				logging.Error(err),
-			)
-			return nil, err
+		if settle != nil && mEvt.margin == nil {
+			// the accounts for the party we need
+			// the accounts for the trader we need
+			mEvt.margin, err = e.GetAccountByID(e.accountID(settle.MarketID, p.Owner, asset, types.AccountTypeMargin))
+			if err != nil {
+				e.log.Error(
+					"Failed to get the party margin account",
+					logging.String("owner-id", p.Owner),
+					logging.String("market-id", settle.MarketID),
+					logging.Error(err),
+				)
+				return nil, err
+			}
 		}
+		// we'll need this account for all transfer types anyway (settlements, margin-risk updates)
+		if mEvt.general == nil {
+			mEvt.general, err = e.GetAccountByID(e.accountID(noMarket, p.Owner, asset, types.AccountTypeGeneral))
+			if err != nil {
+				e.log.Error(
+					"Failed to get the party general account",
+					logging.String("owner-id", p.Owner),
+					logging.String("market-id", settle.MarketID),
+					logging.Error(err),
+				)
+				return nil, err
+			}
+		}
+	} else if mEvt.general == nil {
+		// for the event, the insurance pool acts as the margin/general account
+		mEvt.general = insurance
 	}
 	if p.Type == types.TransferTypeWithdraw || p.Type == types.TransferTypeDeposit {
 		// external account:

--- a/core/settlement/engine.go
+++ b/core/settlement/engine.go
@@ -387,7 +387,7 @@ func (e *Engine) SettleMTM(ctx context.Context, markPrice *num.Uint, positions [
 	}
 	// append wins after loss transfers
 	transfers = append(transfers, wins...)
-	if appendLargest {
+	if len(transfers) > 0 && appendLargest {
 		transfers = append(transfers, largestShare)
 	}
 	e.broker.SendBatch(evts)

--- a/core/settlement/engine.go
+++ b/core/settlement/engine.go
@@ -270,14 +270,15 @@ func (e *Engine) SettleMTM(ctx context.Context, markPrice *num.Uint, positions [
 	e.trades = map[string][]*settlementTrade{} // remove here, once we've processed it all here, we're done
 	evts := make([]events.Event, 0, len(positions))
 	var (
-		largestShare *mtmTransfer       // pointer to whomever gets the last remaining amount from the loss
-		zeroShares   = []*mtmTransfer{} // all zero shares for equal distribution if possible
-		zeroAmts     = false
-		mtmDec       = num.NewDecimalFromFloat(0)
-		lossTotal    = num.UintZero()
-		winTotal     = num.UintZero()
-		lossTotalDec = num.NewDecimalFromFloat(0)
-		winTotalDec  = num.NewDecimalFromFloat(0)
+		largestShare  *mtmTransfer       // pointer to whomever gets the last remaining amount from the loss
+		zeroShares    = []*mtmTransfer{} // all zero shares for equal distribution if possible
+		zeroAmts      = false
+		mtmDec        = num.NewDecimalFromFloat(0)
+		lossTotal     = num.UintZero()
+		winTotal      = num.UintZero()
+		lossTotalDec  = num.NewDecimalFromFloat(0)
+		winTotalDec   = num.NewDecimalFromFloat(0)
+		appendLargest = false
 	)
 
 	// network is treated as a regular party
@@ -348,6 +349,7 @@ func (e *Engine) SettleMTM(ctx context.Context, markPrice *num.Uint, positions [
 				price: markPrice.Clone(),
 			},
 		}
+		appendLargest = true
 	}
 	// no need for this lock anymore
 	e.mu.Unlock()
@@ -385,6 +387,9 @@ func (e *Engine) SettleMTM(ctx context.Context, markPrice *num.Uint, positions [
 	}
 	// append wins after loss transfers
 	transfers = append(transfers, wins...)
+	if appendLargest {
+		transfers = append(transfers, largestShare)
+	}
 	e.broker.SendBatch(evts)
 	timer.EngineTimeCounterAdd()
 	return transfers

--- a/core/settlement/engine_test.go
+++ b/core/settlement/engine_test.go
@@ -752,7 +752,7 @@ func testAddNewPartySelfTrade(t *testing.T) {
 	engine.Update(init)
 	engine.AddTrade(trade)
 	noTransfers := engine.SettleMTM(context.Background(), markPrice, positions)
-	assert.Len(t, noTransfers, 1)
+	assert.Len(t, noTransfers, 2)
 	assert.Nil(t, noTransfers[0].Transfer())
 }
 
@@ -796,7 +796,7 @@ func testAddNewParty(t *testing.T) {
 	engine.Update(init)
 	engine.AddTrade(trade)
 	noTransfers := engine.SettleMTM(context.Background(), markPrice, positions)
-	assert.Len(t, noTransfers, 2)
+	assert.Len(t, noTransfers, 3)
 	for _, v := range noTransfers {
 		assert.Nil(t, v.Transfer())
 	}

